tmt/result.py

```diff
@@ -199,5 +204,5 @@ def to_serialized(self) -> Dict[str, Any]:
         return fields
 
     @classmethod
-    def from_serialized(cls, serialized: Dict[str, Any]) -> 'Result':
-        return cls(serialized, name=serialized.pop('name'))
+    def from_serialized(cls, serialized: ResultData) -> 'Result':
+        return cls(serialized, name=cls.name)
```

Comment:
I believe the original code is correct, see https://tmt.readthedocs.io/en/stable/classes.html#to-serialized-from-serialized-unserialize - `from_serialized()` will be given a dictionary loaded from a YAML file, not a data class instance.